### DISCUSSION TITLE
After a global package has been uninstalled

### DISCUSSION THREAD

**Discussion by parrotdance** [npm/rfcs#277] [Question]
I wrote a package which creates an external config file. I found that when this package is going to be uninstalled, user might forgot to delete this config file, I also have no chance to delete it before/after uninstall.

So the question is: Is there any way (like sorts of hook) to do some cleanup work, before or after the uninstall command?

**Comment by ljharb** [comment]
Seems like the easiest would be to store the config file inside the package’s directory.

What if the user wants to keep the config around, for later reinstalling?

**Reply by parrotdance** [reply]
Oh well, that really is the right way to work around this.

In my case user not need to keep this file for reinstallation.
